src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2258,7 +2258,7 @@ public static bool TrueForAll<T>(T[] array, Predicate<T> match)
             0X7FFFFFC7;
 
         // Private value type used by the Sort methods.
-        private readonly struct SorterObjectArray
+        private readonly ref struct SorterObjectArray
```

Comment:
Is it worth adding a comment about why `ref` exists, here and elsewhere? e.g.
```suggestion
        private readonly ref struct SorterObjectArray // ref isn't functionally necessary but enables additional JIT optimization
```

It's a little unfortunate we're going to start to see this sprinkled around. Is there anything more we could do in the JIT to effectively infer it?

Reply:
Yeah a comment makes sense.

> It's a little unfortunate we're going to start to see this sprinkled around.

I don't see a difference with other things we springle around like marking classes as `sealed`, etc 🤷

Reply:
> I don't see a difference with other things we sprinkle around like marking classes as sealed, etc 🤷

Sealed says "this type wasn't designed to be derived from"; it has use separate from the perf angle. Making these types ref doesn't IMO have any such design benefit; it's being done solely to get a JIT optimization to kick in.

Reply:
Ok, you're right - I don't have a strong opinion on this, I was just testing my script and hoped to see some nice codegen diffs, but they're quite small